core/src/main/java/org/elasticsearch/search/aggregations/pipeline/movavg/models/LinearModel.java

```diff
@@ -41,6 +42,29 @@
     protected static final ParseField NAME_FIELD = new ParseField("linear");
 
     @Override
+    public <T extends Number> double[] predict(Collection<T> values, int numPredictions) {
```

Comment:
I went with the first suggestion, rolling the checks/assertion into `predict()` which then calls down to `doPredict()`.

I couldn't do the second suggestion, because not all the models follow that pattern.  Simple / Linear / Ewma only know how to forecast one value at a time.  E.g. they always return a single value, not an array.  Which also means their prediction code needs to fill the array of predictions with that last value, whereas HL/HW generate new predictions at each point.

If we want less code c/p, going back to the old solution is probably best (concrete `predict()` for simple/linear/ewma and have HL/HW override it).